File: 3955-valid-binary-strings-with-cost-limit/3955-valid-binary-strings-with-cost-limit.cs
public class Solution {
    public IList<string> GenerateValidStrings(int n, int k) {
        List<string> ans = new List<string>();

        void Dfs(int i, int c, bool p, string s) {
            if (c > k) return;

            if (i == n) {
                ans.Add(s);
                return;
            }

            Dfs(i + 1, c, false, s + "0");

            if (!p) {
                Dfs(i + 1, c + i, true, s + "1");
            }
        }

        Dfs(0, 0, false, "");
        return ans;
    }
}

File: 3955-valid-binary-strings-with-cost-limit/3955-valid-binary-strings-with-cost-limit.swift
class Solution {
    func generateValidStrings(_ n: Int, _ k: Int) -> [String] {
        var ans = [String]()

        func dfs(_ i: Int, _ c: Int, _ p: Bool, _ s: String) {
            if c > k { return }

            if i == n {
                ans.append(s)
                return
            }

            dfs(i + 1, c, false, s + "0")

            if !p {
                dfs(i + 1, c + i, true, s + "1")
            }
        }

        dfs(0, 0, false, "")
        return ans
    }
}